### DISCUSSION TITLE
Many to many or it's one to many?

### DISCUSSION THREAD

**Discussion by alexevladgabriel** [prisma/prisma#17922] [Q&A]
Hey, I have a logic problem in my mind. I have the following situation we're `users` should have multiple `clans` and `clan` has mutiple users as members. But clan has a owner what is a user. How could I achieve this?

```prisma
model User {
    id            String    @id @default(cuid())
    name          String?
    email         String?   @unique
    emailVerified DateTime? @map(name: "email_verified")
    image         String?
    createdAt     DateTime  @default(now()) @map(name: "created_at")
    updatedAt     DateTime  @default(now()) @map(name: "updated_at")

    accounts Account[]
    sessions Session[]
    servers  Server[]
    clan     Clan?

    @@map(name: "users")
}
```

```prisma
model Clan {
    id          Int     @id @default(autoincrement())
    slug        String  @unique
    name        String
    tag         String  @unique
    description String?
    image       String?
    site        String?

    ownerId String @unique @map(name: "owner_id")

    server Server[]
    owner  User     @relation(fields: [ownerId], references: [id])

    @@index([ownerId])
    @@map(name: "clans")
}
```

**Comment by ludralph** [comment]
Hi @alexevladgabriel 👋 

Your data model is incorrect for the scenario you described in your message.

To achieve the relationship you described, you should remove the clan field from the User model and instead add a many-to-many relationship between User and Clan using a separate join table, like this:

```prisma
model User {
  id         String     @id @default(cuid())
  name       String?
  email      String?    @unique
  clans      ClanUser[]
  ownedClans Clan[]
}

model Clan {
  id          Int     @id @default(autoincrement())
  slug        String  @unique
  name        String
  tag         String  @unique
  description String?
  image       String?
  site        String?

  owner   User       @relation(fields: [ownerId], references: [id])
  ownerId String
  members ClanUser[]
}

model ClanUser {
  id     Int    @id @default(autoincrement())
  user   User   @relation(fields: [userId], references: [id])
  userId String
  clan   Clan   @relation(fields: [clanId], references: [id])
  clanId Int

  @@unique([userId, clanId])
}
```
In this model, the User model has a many-to-many relationship with the Clan model through the ClanUser model. The ClanUser model acts as a join table between User and Clan, with the foreign keys userId and clanId.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
